### PR TITLE
commands v2

### PR DESCRIPTION
**Pull request by nxtdaydelivery**
## Description
commands v2 yay
p.s. this is breaking

## Related Issue(s)
commands not working

## Checklist
- [x] I made a clear description of what was changed
- [X] I stated why these changes were necessary
- [X] I updated documentation or said what needs to be updated
- [!] I made sure these changes are backwards compatible
- [X] This pull request is for one feature/bug fix

### CODE REVIEW

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:51]: I feel like descriptor should be something like @ Main here instead and value shouldn't be needed here

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:56]: This should be something like @ SubCommand

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:61]: This could be something like @SubCommandGroup since command implies it is top level

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:63]: also @ Subcommand here

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:76]: also @ SubCommand here

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:86]: also @ SubCommand here

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:97]: also @ SubCommand here

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:107]: also @ SubCommand here

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (DeDiamondPro) [inline src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:87]: @ Descriptor could be replaced by @ Description (here and in other places like it)

**Review by DeDiamondPro — COMMENTED** [review]
Overall I think its an improvement but the annotation names are kind of confusing

File: src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java
Comment (nxtdaydelivery) [reply src/main/java/cc/polyfrost/oneconfig/internal/command/OneConfigCommand.java:51]: they don't have to be annotated, its called @ Descriptor because it adds info to the help message. value is displayed in the help message as well if its present.

File: src/main/java/cc/polyfrost/oneconfig/config/annotations/Exclude.java
Comment (nxtdaydelivery) [inline src/main/java/cc/polyfrost/oneconfig/config/annotations/Exclude.java:34]: ps. this is a rollback

**Review by DeDiamondPro — APPROVED** [review]
For obvious reasons don't merge pr yet

**Review by Wyvest — DISMISSED** [review]

File: src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java
Comment (Wyvest) [inline src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java:59]: why???? we had argument parsers for a reason?

File: src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java
Comment (Wyvest) [inline src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java:37]: can we name this something else

**Review by Wyvest — CHANGES_REQUESTED** [review]
looks good overall, but some weird decisions in names and stuff

File: src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java
Comment (nxtdaydelivery) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java:37]: sure this was diamonds idea

Comment (xtrm-en) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java:37]: how about `SubGroup`, `Subcommand`, or simply `Sub`? @Wyvest

File: src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java
Comment (xtrm-en) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java:59]: `description` is for a more precise optional description for error display
`autoCompletesTo` covers more cases than arg parsers (example: String argparser cant know we want only profile names for example)

Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java:59]: > `autoCompletesTo` covers more cases than arg parsers (example: String argparser cant know we want only profile names for example)

it can know???

Comment (xtrm-en) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java:59]: uhhhh octopus

File: src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java
Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/SubCommandGroup.java:37]: > how about `SubGroup`, `Subcommand`, or simply `Sub`? @Wyvest

actually this is fine

File: src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java
Comment (Wyvest) [reply src/main/java/cc/polyfrost/oneconfig/utils/commands/annotations/Description.java:59]: can you remove it please ty . . .

**Review by Wyvest — APPROVED** [review]

**Comment by CaledonianEH** [comment]
Must fix the API Check so it can properly build

The QC check will be ignored for this PR, and QC will be left to the developer's judgment.

From my review, it looks fine, and I will approve it, but don't trust my judgement

**Review by CaledonianEH — DISMISSED** [review]
👍

**Review by xtrm-en — APPROVED** [review]
:+1:

**Review by Wyvest — CHANGES_REQUESTED** [review]
can we readd argument parser's abilities to add tab completion

**Review by xtrm-en — APPROVED** [review]

**Review by Wyvest — APPROVED** [review]

**Comment by Wyvest** [comment]
Will merge once our maven repo is fully set up

**Pull request merged**
